replica/table.cc

```diff
@@ -734,6 +735,25 @@ table::try_flush_memtable_to_sstable(lw_shared_ptr<memtable> old, sstable_write_
     co_return co_await with_scheduling_group(_config.memtable_scheduling_group, std::ref(try_flush));
 }
 
+future<> table::maybe_wait_for_sstable_count_reduction() {
+    while (true) {
+        if (_async_gate.is_closed()) {
+            co_return;
+        }
+        auto sstable_runs_with_memtable_origin = boost::copy_range<std::unordered_set<utils::UUID>>(
+            *_main_sstables->all()
+            | boost::adaptors::filtered([] (const sstables::shared_sstable& sst) {
+                return sst->get_origin() == "memtable";
+            })
+            | boost::adaptors::transformed(std::mem_fn(&sstables::sstable::run_identifier)));
+        if (sstable_runs_with_memtable_origin.size() < schema()->max_compaction_threshold() || is_auto_compaction_disabled_by_user()) {
+            co_return;
+        }
+        trigger_compaction();
```

Comment:
Do we really need to trigger compaction here? Any sstable creation should have already triggered compaction, so presumably we're just waiting for compaction to catch up.

OTOH this may help to overcome a bug where we "forget" to trigger compaction.

Reply:
Yes, it's not strictly needed, but I suggested it as trigger_compaction() is idempotent and we want to reinforce here the need for compaction, to reduce the chances of falling into an endless wait, if compaction wasn't scheduled for the table due to a problem.

Reply:
A comment here then.